kobotoolbox/kpi
# feat(organizations): update organizations endpoint to block post and delete requests TASK-964

### 🗒️ Checklist

1. [x] run linter locally
2. [ ] update all related docs (API, README, inline, etc.), if any
3. [x] draft PR with a title `<type>(<scope>)<!>: <title> TASK-1234`
4. [x] tag PR: at least `frontend` or `backend` unless it's global
5. [x] fill in the template below and delete template comments
6. [x] review thyself: read the diff and repro the preview as written
7. [x] open PR & confirm that CI passes
8. [ ] request reviewers, if needed
9. [ ] delete this section before merging

### 📣 Summary
Restricted POST and DELETE methods for `organizations` endpoint to prevent unintended changes.


### 📖 Description
Previously, the organizations endpoint allowed POST and DELETE requests, which could lead to accidental changes or deletions. This update restricts these methods, ensuring that organizations data remains intact and secure.


### 👀 Preview steps
- Send a GET request to the organizations endpoint to retrieve a list of organizations.
- Attempt to send a POST request to create a new organization (this should fail).
- Attempt to send a DELETE request to delete an existing organization (this should fail).
- Send a PATCH request to update an existing organization (this should succeed).


### 💭 Notes
- This change is a security enhancement to prevent accidental or malicious modifications to organizations data.
- Tests have been updated to reflect the new behaviour.
